aws-dual-stack/openshift-upgrade-aws.sh

```diff
@@ -0,0 +1,238 @@
+               --ipv6-cidr-block-network-border-group $AWS_DEFAULT_REGION \
+               --vpc-id $VPC_ID )
+  echo $CIDER_ADDED
+  if [[ "$CIDER_ADDED" == *"error"* ]]; then
+    echo "unable to assign ipv6 block"
+    exit 1
+  fi
+  VPC_SUBNET_BLOCK=$(getVPCIPV6SubnetBlock $VPC_ID)
+  if [ "$VPC_SUBNET_BLOCK" == false ]; then
+    echo "unable to assign ipv6 block"
+    exit 1
+  fi
+fi
+
+################################# SUBNETS
+echo "VPC IPV6 Subnet Block: $VPC_SUBNET_BLOCK"
+
+VPC_SUBNET_ADDRESS=$(echo $VPC_SUBNET_BLOCK | sed 's/\/56//')
+echo "Using VPC subnet address block $VPC_SUBNET_ADDRESS"
+
+#generate 16 subnets from the VPC ipv6 block of size /64  4 bits between 60 to 64
+IFS=$'\n' IPV6SUBNETS=($(sipcalc -S 64 $VPC_SUBNET_ADDRESS/60  -u | grep "Expanded Address" | awk '{print $4}'))
+echo "Generated ${#IPV6SUBNETS[@]} subnets"
+
+AWS_SUBNET_DESC=$(aws ec2 describe-subnets \
+    --filters "Name=vpc-id,Values=$VPC_ID" )
+
+echo $AWS_SUBNET_DESC | jq -r '.Subnets[] | select(.Ipv6CidrBlockAssociationSet[].Ipv6CidrBlockState.State=="associated") | .SubnetId'
+
+IFS=$'\n' AWS_SUBNETS=($(echo $AWS_SUBNET_DESC | jq -r '.Subnets[].SubnetId'))
+echo "Found subnets in VPC: ${AWS_SUBNETS[@]}"
+echo "---------------------------"
+num_ipv6_subnets=${#IPV6SUBNETS[@]}
+num_aws_subnets=${#AWS_SUBNETS[@]}
+
+if [ "$num_ipv6_subnets" -le "$num_aws_subnets" ]; then
+  echo "Test failed: This script assumed Openshift created 16 subnets by default"
+  echo "Not enough subnets: $num_ipv6_subnets <= $num_aws_subnets"
+  exit 1
+fi
+
+for (( j=0; j<${num_aws_subnets}; j++ ));
+do
+  SUBNET_ID="${AWS_SUBNETS[$j]}"
+
+  AWS_SUBNETS_W_IPV6=$(echo $AWS_SUBNET_DESC | jq -r '.Subnets[] | select(.Ipv6CidrBlockAssociationSet[].Ipv6CidrBlockState.State=="associated") | .SubnetId' | grep "$SUBNET_ID" || true)
+
+  if [ -z "$AWS_SUBNETS_W_IPV6" ]; then
+    printf "Adding ipv6 block: %s to %s\n"  "${IPV6SUBNETS[$j]}/64" "$SUBNET_ID"
+    subnet_applied=$(aws ec2 associate-subnet-cidr-block --subnet-id $SUBNET_ID --ipv6-cidr-block "${IPV6SUBNETS[$j]}/64")
+    echo $subnet_applied
+  else
+    echo "Skipping SubnetID: $AWS_SUBNETS_W_IPV6"
+  fi
+
+done
+
+echo "---------------------------"
+################################# route-tables public
+ROUTE_TABLES=$(aws ec2 describe-route-tables --filters "Name=vpc-id,Values=$VPC_ID" --query 'RouteTables[?Tags[?Key==`Name` && contains(Value,`public`)]]' )
+ROUTE_TABLE_ID=$(echo $ROUTE_TABLES | jq -r '.[].RouteTableId')
+if [ -z "$ROUTE_TABLE_ID" ];then
+  echo "Public ROUTE_TABLE not found"
+  exit 1
+fi
+
+INTERNET_GATEWAYID=$(aws ec2 describe-internet-gateways --filters "Name=attachment.vpc-id,Values=$VPC_ID" | jq -r '.InternetGateways[].InternetGatewayId')
+if [ -z "$INTERNET_GATEWAYID" ];then
+  echo "Public INTERNET_GATEWAYID not found"
+  exit 1
+fi
+
+ROUTE_TABLE_CHECK=$(echo $ROUTE_TABLES | grep "$INTERNET_GATEWAYID" | wc -l)
+if [ "$ROUTE_TABLE_CHECK" == 0 ]; then
```

Comment:
nit: What does bash do if you do equality comparisons with "$var"? I thought it would put them in quotes too (e.g. "0"). Is that only with `[[ ]]`

Reply:
I believe it only adds quotes for double brackets